ec2launch/windows-setup.tpl.yml

```diff
@@ -1,23 +1,27 @@
 ---
 version: 1.0
 tasks:
-  # Set up the Samba share
+  # Expand the OS partition and set up the Samba share
   - task: executeScript
     inputs:
-      - frequency: once
+      - frequency: always
         type: powershell
         runAs: localSystem
         content: |-
-          $samba_server = ("${samba_server_input}" -split ",")[0]
+          $DiskPath = (Get-Partition -DriveLetter C).DiskPath
+          if ((Get-Partition -DiskId $DiskPath).Count -gt 2) {
+            "select disk 0`nselect partition 2`nextend`n" | diskpart.exe
```

Comment:
Might be worth some additional comments here about magic drive letter `C` and magic partition `2`, unless you want to make those input variables.  Even then, there should probably still be a comment.

Reply:
I agree with @dav3r.

Reply:
I agree on the partition number, but `C` is the standard OS partition drive letter for Windows.

Reply:
I get that, but even so, it doesn't hurt to mention in a comment that:
```
# C is the standard OS partition drive letter for Windows
```

Reply:
Well I was more referencing that we use a lot of Linux paths without explanation x)

Reply:
I don't know if that's exactly analogous, but I leave it up to you unless @jsf9k feels strongly about this.

Reply:
I don't have any issue adding a comment explaining. My point was more it's worth commenting because Windows-related knowledge is not strongly in the domain knowledge of most of our team, but we use a lot of Linux-specific things without comment under the assumption that _it is_ in the domain knowledge of our team.

Reply:
That's fair and I'm all for us doing a better job of adding comments for similar types of Linux thangs.

Reply:
I actually changed the `Get-Partition` call to be consistent with the commands we are piping to `diskpart.exe` in 1ccee97. I did add comments to overall explain things in the Windows instance userdata in 277ec3c.

Reply:
Those comments look great to me - thanks!